snuba/query/processors/readonly_events.py

```diff
@@ -0,0 +1,37 @@
+from snuba import state
+from snuba.datasets.schemas.tables import TableSource
+from snuba.query.query import Query
+from snuba.query.query_processor import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+
+class ReadOnlyTableSelector(QueryProcessor):
+    """
+    Replaces the data source in the query with a TableSource if the table
+    name of the original datasource is the one provided to the constructor.
```

Comment:
This doc doesn't seem 100% accurate, there are some other conditions where the dataset is not swapped?

Reply:
will change